### PR TITLE
Animated marker added to Symbology

### LINKED ISSUE #7468: New marker symbol type for "Animated Marker" (Request in QGIS)
### Request for documentation
From pull request QGIS/qgis#48060
Author: @nyalldawson
QGIS version: 3.26

**New marker symbol type for "Animated Marker"**

### PR Description:
This new marker symbol type allows points to be rendered using an animated marker, sourced from an animated gif, webp or mng animation.

Options are present for marker file, size, angle and frame rate.

There are two ways in which animated symbols are handled:

1. If the map itself is considered an animation, then the frame rendered for the animated marker is based on the map animation
frame and frame rate. This is the case when the temporal controller is set to the Animation mode. In this case the
animated markers will follow the temporal controller animation, e.g. pausing when the animation is paused, advancing frames
with the animation, etc. The map will also be redrawn using the frame rate set for the temporal animation.

This mode also applies when exporting an animation from the temporal controller.

It's also the mode used when a plugin specifically sets the frame rate and current frame QgsMapSettings properties, so
e.g. @timlinux's QGIS Animation Workbench plugin will dictate the marker animation frame to render.

2. If the map is NOT considered an animation (i.e. it's just a plain old normal QGIS project), then the frame to render will be based on the current timestamp alone.

Markers will be animated when their corresponding layer is set to a temporal mode, and a temporal animation is playing.

### Animated marker in a standard map, no temporal animation settings

Peek 2022-04-03 07-56

### Animated marker when used with temporal controller in animation mode

Peek 2022-04-03 07-59

### Commits tagged with [need-docs] or [FEATURE]
"[feature] New marker symbol type for \"Animated Marker\"\n\nThis new marker symbol type allows points to be rendered using\nan animated marker, sourced from an animated gif, webp or mng\nanimation.\n\nOptions are present for marker file, size, angle and frame rate.\n\nThere are two ways in which animated symbols are handled:\n\n1. If the map itself is considered an animation, then the frame\nrendered for the animated marker is based on the map animation\nframe and frame rate. This is the case when the temporal\ncontroller is set to the Animation mode. In this case the\nanimated markers will follow the temporal controller animation,\ne.g. pausing when the animation is paused, advancing frames\nwith the animation, etc. The map will also be redrawn using\nthe frame rate set for the temporal animation.\n\nThis mode also applies when exporting an animation from the\ntemporal controller.\n\nIt's also the mode used when a plugin specifically sets the\nframe rate and current frame QgsMapSettings properties, so\ne.g. @timlinux's QGIS Animation Workbench plugin will dictate\nthe marker animation frame to render.\n\n2. If the map is NOT considered an animation (i.e. it's just\na plain old normal QGIS project), then the frame to render\nwill be based on the current timestamp alone.\n\nMarkers will be animated when their corresponding layer is set\nto a temporal mode, and a temporal animation is playing."

### PR DESCRIPTION
**Pull request by selmaVH1**
fixes #7468 
Animated marker added to the Style library.

### CODE REVIEW

File: docs/user_manual/style_library/symbol_selector.rst [hunk 1]
```diff
@@ -820,6 +821,38 @@ The magnitude of field can be scaled up or down to an appropriate size for
 viewing the field.
 
 
+.. _animated_marker:
```
Comment (DelazJ) [inline docs/user_manual/style_library/symbol_selector.rst:824]:
```suggestion
.. index:: Animation
.. _animated_marker:
```
Let's connect sections in the index page also

**Comment by selmaVH1** [comment]
> @selmaVH1 I think we need to showcase this type of symbol and an animation would probably be appropriate... maybe the first one in the issue report?

@DelazJ make sense, I will add animation and go through other suggestions.

**Comment by selmaVH1** [comment]
@DelazJ can you, please review this PR again? I added both animations, they can be helpful. I have also removed paragraph titles, so now everything is in one paragraph, but I'm still not sure is it clear enough for users. 
Another thing, I added screenshots for PDF, but I can't create preview for PDF so I'm not sure is it ok...

File: docs/user_manual/style_library/symbol_selector.rst [hunk 2]
```diff
@@ -820,6 +821,51 @@ The magnitude of field can be scaled up or down to an appropriate size for
...
+...............
+
+Animated marker symbol type allows you to use a :file:`.GIF`, :file:`.WebP`,
+:file:`.MNG`, etc. animation file to represent points on your map. 
+You can specify:
+
+* :guilabel:`File` path,
+* :guilabel:`Frame rate`: number of steps that are shown per second,
+  indicating how fast the animation is played,
+* :guilabel:`Size` in any :ref:`supported unit <supported_units>`,
```
Comment (DelazJ) [inline docs/user_manual/style_library/symbol_selector.rst:837]:
```suggestion
* :guilabel:`Size` in any :ref:`supported unit <unit_selector>`,
```
oups, sorry!

File: docs/user_manual/style_library/symbol_selector.rst [hunk 3]
```diff
@@ -820,6 +821,38 @@ The magnitude of field can be scaled up or down to an appropriate size for
...
+.. _animated_marker:
+
+Animated marker
+...............
+
+Animated marker symbol type allows you to use animated markers, which can be
+GIFs, WebP, or MNG animations, to represent points on your map. You 
+can specify marker ``file``, ``size``, ``angle``, and ``frame rate``.
+
+There are two ways in which animated symbols are handled:
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/user_manual/style_library/symbol_selector.rst]: > I have also removed paragraph titles, so now everything is in one paragraph, but I'm still not sure is it clear enough for users.

Indeed. Maybe do we need to keep this sentence?

File: docs/user_manual/style_library/symbol_selector.rst [hunk 4]
```diff
@@ -820,6 +821,51 @@ The magnitude of field can be scaled up or down to an appropriate size for
...
+
+**When your map is not configured as an animation** (i.e. it's a standard QGIS 
+project without animations), the frame for the animated markers will be 
+determined solely by the current timestamp. 
+
+.. figure:: /docs/user_manual/style_library/img/animated_marker_map.*
+   :align: center
+   :width: 100%
+
+   Animated marker when map is not configured as animation
```
[the remark this answers is not in the document]
Comment (DelazJ) [reply docs/user_manual/style_library/symbol_selector.rst]: 😮 svg? I should stop review when too late..

File: docs/user_manual/style_library/symbol_selector.rst [hunk 5]
```diff
@@ -820,6 +821,57 @@ The magnitude of field can be scaled up or down to an appropriate size for
...
+
+There are two ways to handle animated symbols:
+
+**When your map is not configured as an animation** (i.e. it's a standard QGIS 
+project without animations), the frame for the animated markers will be 
+determined solely by the current timestamp. 
+
+.. only:: html
+
+.. figure:: /docs/user_manual/style_library/img/animated_marker_map.gif
```
Comment (DelazJ) [inline docs/user_manual/style_library/symbol_selector.rst:852]: The whole block with caption should be indented, under the `only...` block. And use relative path to file. See my previous suggestion

**Pull request merged**